main.py

```diff
@@ -56,26 +68,64 @@
 
 if train:
 	# Generate datasets
-	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config)
+	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config,random_split=random_split, disjoint_split=disjoint_split)
 
 	# Initialize final model
-	model = Model(config=config)
+	if use_semantic_embeddings:
+		Sy_word = []
+		with open(os.path.join(config.folder, "pretraining", "words.txt"), "r") as f:
+			for line in f.readlines():
+				Sy_word.append(line.rstrip("\n"))
+		glove_embeddings=obtain_glove_embeddings(semantic_embeddings_path, Sy_word )
+		model = Model(config=config,pipeline=False, use_semantic_embeddings = use_semantic_embeddings, glove_embeddings=glove_embeddings)
```

Comment:
Very minor, but you can just have `use_semantic_embeddings = True` here and in the clause below.
Doesn't affect the accuracy of this code so you can just ignore this message

Reply:
Makes sense. I think I will need to refactor code a bit later.